app/src/org/commcare/models/database/connect/ConnectDatabaseUpgrader.java

```diff
@@ -326,6 +333,44 @@ private boolean upgradeSixSeven(SQLiteDatabase db) {
         return addTableForNewModel(db, ConnectPaymentUnitRecord.STORAGE_KEY, new ConnectPaymentUnitRecord());
     }
 
+    private boolean upgradeSevenEight(SQLiteDatabase db) {
+        db.beginTransaction();
+
+        try {
+            //First, migrate the old ConnectJobRecord in storage to the new version
+            db.execSQL(DbUtil.addColumnToTable(
+                    ConnectJobRecord.STORAGE_KEY,
+                    ConnectJobRecord.META_USER_SUSPENDED,
+                    "TEXT"));
+
+
+            SqlStorage<Persistable> oldStorage = new SqlStorage<>(
+                    ConnectJobRecord.STORAGE_KEY,
+                    ConnectJobRecordV5.class,
+                    new ConcreteAndroidDbHelper(c, db));
+
+            SqlStorage<Persistable> newStorage = new SqlStorage<>(
+                    ConnectJobRecord.STORAGE_KEY,
+                    ConnectJobRecord.class,
+                    new ConcreteAndroidDbHelper(c, db));
+
+            for (Persistable r : oldStorage) {
+                ConnectJobRecordV5 oldRecord = (ConnectJobRecordV5)r;
+                ConnectJobRecord newRecord = ConnectJobRecord.fromV5(oldRecord);
+                //set this new record to have same ID as the old one
+                newRecord.setIsUserSuspended(oldRecord.getIsUserSuspended());
```

Comment:
It's important that this line set the ID of the new entry to match the ID of the old entry (note the setID and getID functions are part of the Persisted class for accessing the actual DB IDs, not accessors to custom fields on our ConnectJobRecord).

This line MUST be:
newRecord.setID(oldRecord.getID());

We handle setting the isUserSuspended flag inside the fromV5 function (defaulting to false).